loopy/isl_helpers.py

```diff
@@ -60,7 +60,30 @@ def dump_space(ls):
 
 # {{{ make_slab
 
-def make_slab(space, iname, start, stop):
+def make_slab(space, iname, start, stop, step=1):
+    """
+    Returns an instance of :class:`islpy._isl.BasicSet`, which satisfies the
+    constraint ``start <= step*iname < stop``.
```

Comment:
@kaushikcfd This one is dangerous, and I think it's pretty close to a showstopper. I vote that we fix this before merging. `slice` for example does not work like this, and I think this behavior is highly unintuitive.
```
>>> a = list(range(30))
>>> a[3:40:3]
[3, 6, 9, 12, 15, 18, 21, 24, 27]
```
vs.
```
>>> import islpy as isl
>>> s = isl.BasicSet("{[i]: 1=0}"); s = make_slab(s.space, "i", 3, 40, 3); s
BasicSet("{ [i] : 0 < i <= 13 }")
```
Do we have any tests exercising strided array access? (I'm guessing/hoping not.)

One potential way out of this is to ban strides for the time being and revert this function.

Reply:
#396  (Thanks @kaushikcfd!)